src/tokenizer.rs

```diff
@@ -1994,6 +2023,70 @@ impl<'a> Tokenizer<'a> {
         )
     }
 
+    /// Reads a quote delimited string without "backslash escaping" or a word
+    /// depending on whether `chars.next()` delivers a `'`.
+    ///
+    /// See <https://docs.oracle.com/en/database/oracle/oracle-database/19/sqlrf/Literals.html>
+    fn tokenize_word_or_quote_delimited_string(
+        &self,
+        chars: &mut State,
+        // the prefix that introduced the possible literal or word,
+        // e.g. "Q" or "nq"
+        word_prefix: &[char],
+        // turns an identified quote string literal,
+        // ie. `(start-quote-char, string-literal, end-quote-char)`
+        // into a token
+        as_literal: fn(char, String, char) -> Token,
```

Comment:
since the function is only used twice, I think it would be reasonable to have the caller do something like

```rust
chars.next(); // consume 'q'
if chars.peek() != Some('\'') {
    return 'q' + tokenize_word()
}
let QuoteString(quote_delimited_string())
```
i.e. that this function should only care about parsing the string, in order to simplify its implementation